### PR TITLE
add the readme/tutorial and refactor i2c code

### PR DESCRIPTION
**Pull request by oli-obk**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -9,6 +9,8 @@ cortex-m = "0.1.4"
 r0 = "0.1.0"
 rlibc = "1.0.0"
 volatile = "0.2.0"
+# used for i2c
+byteorder = { version = "0.5.3", default-features = false, path = "../byteorder" }
```
Comment (phil-opp) [inline Cargo.toml]: Is this `path` dependency on purpose?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+-  install a nightly compiler
+    * `rustup update nightly`
+-  dowload the rust source code
+    * if your rustup does not have the `component` subcommand: `rustup self update`
+    * `rustup component add rust-src --toolchain nightly`
+-  install `xargo`
+    * `rustup run nightly cargo install xargo`
+    * NOTE: do **not** run this command in the `novemb-rs-stm32f7` folder, you will get errors about the compiler not finding the standard library
+-  get the demo code
+    * `git clone https://github.com/phil-opp/novemb-rs-stm32f7.git`
```
Comment (phil-opp) [inline README.md]: The repo is owned by the embed-rs organization now

File: src/i2c.rs [hunk 3]
```diff
@@ -109,130 +99,166 @@ fn icr_clear_all() -> i2c1::Icr {
...
-        cr2.set_autoend(false); // automatic_end_mode
-        self.0.cr2.write(cr2);
-
-        // write high address byte to transmit data register
-        try!(self.wait_for_txis());
-        self.0.txdr.update(|r| r.set_txdata((register_address >> 8) as u8)); // transmit_data
+#[derive(Debug, Clone, Copy)]
+pub enum Address {
+    U7(u8),
+    U10(u16),
```
Comment (phil-opp) [inline src/i2c.rs]: What does the `U` prefix stand for?  (just curious)

File: src/touch/mod.rs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+pub fn init_ft6x06(i2c_3: &mut i2c::I2C) -> Result<(), i2c::Error> {
+    // reset device
+    i2c_3.write(FT6X06_ADDRESS, 0, 0)?;
+    // read and check device family ID
+    assert_eq!(i2c_3.read(FT6X06_ADDRESS, 0xA8)?, 0x11);
+    // set polling mode
+    i2c_3.write(FT6X06_ADDRESS, 0xA4, 0)?;
+    Ok(())
+}
```
Comment (phil-opp) [inline src/touch/mod.rs]: We should mention this new touch module in the PR or commit descriptions. Alternatively we could open a separate PR for it.

File: Cargo.toml (hunk 1 above)
Comment (oli-obk) [inline Cargo.toml]: whoops

File: src/i2c.rs (hunk 3 above)
Comment (oli-obk) [inline src/i2c.rs]: follows the `u8`, `u16` naming scheme

File: src/touch/mod.rs (hunk 4 above)
Comment (oli-obk) [inline src/touch/mod.rs]: oh sorry, it's commented out anyway

**Comment by oli-obk** [comment]
closed in favour of #2

**Pull request closed without merging**